### PR TITLE
Rollback integration test

### PR DESCRIPTION
**Pull request by Tolsi**

### CODE REVIEW

File: src/it/scala/com/wavesplatform/it/RollbackSpec.scala [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+import scala.collection.mutable
+import scala.concurrent.Await.result
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.Future.traverse
+import scala.concurrent.duration._
+import scala.collection.JavaConverters._
+
+class RollbackSpec extends FreeSpec with ScalaFutures with IntegrationPatience
```
Comment (ismagin) [inline src/it/scala/com/wavesplatform/it/RollbackSpec.scala]: Basically the difference between two tests are
```
 +      _ <- traverse(nodes)(_.rollback(startHeight, returnToUTX = false))
 +      _ <- processRequests(requests)
```
vs 
```
 +      _ <- traverse(nodes)(_.rollback(startHeight))
```
Might be reasonalbe to refactor that.

File: src/it/scala/com/wavesplatform/it/RollbackSpec.scala [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  with Matchers with TransferSending with IntegrationNodesInitializationAndStopping {
+  override val docker = new Docker()
+  private val allNodes = Docker.NodeConfigs.getConfigList("nodes").asScala
+  // there are nodes with big and small balances to reduce the number of forks
+  override val nodes = Seq(allNodes.head, allNodes.last).map(docker.startNode)
+
+  "Apply the same transfer transactions twice with return to UTX" in {
```
Comment (ismagin) [inline src/it/scala/com/wavesplatform/it/RollbackSpec.scala]: Do we apply the same transactions twice?

File: src/it/scala/com/wavesplatform/it/RollbackSpec.scala [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      requests = generateRequests(301, b)
+      startHeight <- Future.traverse(nodes)(_.height).map(_.min)
+      _ <- processRequests(requests)
+      hashAfterFirstTry <- traverse(nodes)(_.waitForDebugInfoAt(startHeight + waitBlocks).map(_.stateHash)).map(infos => {
+        all(infos) shouldEqual infos.head
+        infos.head
+      })
+      stateAfterFirstTry <- nodes.head.debugStateAt(startHeight + waitBlocks)
+      _ <- traverse(nodes)(_.rollback(startHeight, returnToUTX = false))
+      _ <- processRequests(requests)
```
Comment (ismagin) [inline src/it/scala/com/wavesplatform/it/RollbackSpec.scala]: is there a way we can just rollback and observe that transactions are NOT there?

File: src/it/scala/com/wavesplatform/it/TransferSending.scala [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  def nodes: Seq[Node]
+
+  def generateRequests(n: Int, balances: mutable.Map[String, Long]): Seq[Req] = {
+    val addresses = nodes.map(_.address)
+    val sourceAndDest = (1 to n).map { _ =>
+      val Seq(src, dest) = Random.shuffle(addresses).take(2)
+      (src, dest)
+    }
+    val (_, requests) = sourceAndDest.foldLeft((balances, ArrayBuffer.empty[Req])) {
+      case ((b, reqs), (src, dest)) =>
```
Comment (ismagin) [inline src/it/scala/com/wavesplatform/it/TransferSending.scala]: what is the point of making fold if the collection is mutable? 
Either make it immutable or just do `map` here if you wish to keep this mutable.

File: src/it/scala/com/wavesplatform/it/RollbackSpec.scala (hunk 2 above)
Comment (Tolsi) [inline src/it/scala/com/wavesplatform/it/RollbackSpec.scala]: yes

File: src/it/scala/com/wavesplatform/it/RollbackSpecSuite.scala [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      stateAfterFirstTry should contain theSameElementsAs stateAfterSecondTry
+      hashAfterFirstTry shouldBe hashAfterSecondTry
+    }, 5.minutes)
+  }
+
+  "Apply the same transfer transactions twice with return to UTX" in {
+    rollbackTest((startHeight, _) => traverse(nodes)(_.rollback(startHeight)))
+  }
+
+  "Apply the same transfer transactions twice without return to UTX" in {
```
Comment (ismagin) [inline src/it/scala/com/wavesplatform/it/RollbackSpecSuite.scala]: This test checks nothing, please remove

**Pull request merged**